packages/shylu/shylu_dd/frosch/test/Stratimikos/main.cpp

```diff
@@ -0,0 +1,215 @@
+            }*/
+            
+            if (Comm->MyPID()==0) cout << "ASSEMBLY...";
+            
+            ParameterList GalerList;
+            GalerList.set("nx", N*M);
+            GalerList.set("ny", N*M);
+            GalerList.set("nz", N*M);
+            GalerList.set("mx", N);
+            GalerList.set("my", N);
+            GalerList.set("mz", N);
+            
+            RCP<Epetra_Map> UniqueMap;
+            RCP<Epetra_Map> RepeatedMap;
+            RCP<Epetra_CrsMatrix> K;
+            
+            if (Dimension==2) {
+                UniqueMap.reset(Galeri::CreateMap("Cartesian2D", *Comm, GalerList));
+                K.reset(Galeri::CreateCrsMatrix("Laplace2D", UniqueMap.get(), GalerList));
+            } else if (Dimension==3) {
+                UniqueMap.reset(Galeri::CreateMap("Cartesian3D", *Comm, GalerList));
+                K.reset(Galeri::CreateCrsMatrix("Laplace3D", UniqueMap.get(), GalerList));
+            }
+            
+            EpetraCrsMatrixT<GO,NO> xK(K);
+            RCP<CrsMatrix<SC,LO,GO,NO> > xCrsMat = rcpFromRef(xK);
+            RCP<Matrix<SC,LO,GO,NO> > xMat = rcp(new CrsMatrixWrap<SC,LO,GO,NO>(xCrsMat));
+            xMat->describe(*fancy,Teuchos::VERB_EXTREME);
+            if (Comm->MyPID()==0) cout << "done" <<std::endl;
+            
+            RCP<Epetra_MultiVector> solution(new Epetra_MultiVector(*UniqueMap,1));
+            EpetraMultiVectorT<GO,NO> eSolution(solution);
+            RCP<MultiVector<SC,LO,GO,NO> > xSolution = rcpFromRef(eSolution);
+
+            
+            
+            RCP<Epetra_MultiVector> rightHandSide(new Epetra_MultiVector(*UniqueMap,1));
+            EpetraMultiVectorT<GO,NO> eRightHandSide(rightHandSide);
+            RCP<MultiVector<SC,LO,GO,NO> > xRightHandSide = rcpFromRef(eRightHandSide);
+            
+            xSolution->putScalar(0.0);
+            xRightHandSide->putScalar(1.0);
+            
+            RCP<const Map<LO,GO,NO> > map = xMat->getRowMap();
+            Comm->Barrier();
+            
+            fancyout<<"-------Create Null Space----------\n";
+            
+            RCP<const Thyra::LinearOpBase<SC> > thyraA = Xpetra::ThyraUtils<SC,LO,GO,NO>::toThyra(xCrsMat
+                                                                                                  );
+            RCP<Thyra::MultiVectorBase<SC> >thyraX = Teuchos::rcp_const_cast<Thyra::MultiVectorBase<SC> >(Xpetra::ThyraUtils<SC,LO,GO,NO>::toThyraMultiVector(xSolution));
+            RCP<const Thyra::MultiVectorBase<SC> >thyraB = Xpetra::ThyraUtils<SC,LO,GO,NO>::toThyraMultiVector(xRightHandSide);
+            
+            Comm->Barrier();
+            
+            fancyout<<"-------To Thyra----------\n";
+            //Stratimikos
+            Stratimikos::DefaultLinearSolverBuilder linearSolverBuilder;  // This is the Stratimikos main class (= factory of solver factory).
+            Stratimikos::enableFROSchTwoLevel<LO,GO,NO>(linearSolverBuilder);
```

Comment:
Do we always have to enable FROSch? If I am not mistaken, Ifpack and MueLu are always enabled, right?

Reply:
It's certainly possible to enable Stratimikos but disable MueLu.

Reply:
I mean that we should make sure that FROSch is enabled by default in Stratimikos as long as the package FROSch is enabled. Isn't this the case for MueLu and Ifpack?

Reply:
Just use the macros that Stratimikos should define, to test whether MueLu and Ifpack are enabled.  Code needs to build _always_, not just with default settings.

Reply:
@mhoemmen Ok, I think I mixed some things up. I was just wondering if there is any difference between the implementation for FROSch and MueLu.